### ISSUE TITLE
File Browser takes all mouse click events on Ubuntu

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#584]
_[Trac time 20090128 192656Z]_ From Christopher Armstrong on the peppy mailing list:

> Okay, this one is truly nasty. It may be specific to my window
> manager, which is Compiz (default on Ubuntu for a while). When I click
> on the File Browser gutter-button, it pops up... and if I click
> outside of the popup, it disappears. Here's the bug: now, no matter
> where I click on the screen (really!) it pops up again and the click
> doesn't go where I want it to. If I click anywhere again, it closes.
> So my mouse has become useless, and alt+tab doesn't get me away from
> Peppy (to, e.g., a terminal to kill peppy :). Fortunately the
> keybinding I have mapped to close a window still works, so I can close
> peppy.
> 
> This is happening on the very latest trunk on Ubuntu 8.10.

**Comment by robmcmullen** [comment]
_[Trac time 20090128 192803Z]_ I was able to duplicate it.  Interestingly I thought it would also
apply to the other springtabs, but it's only the File Browser one.
The Code Explorer and Projects plugin work fine...  So, there's some
interaction with the wxGenericDirCtrl and the springtab...

I get this traceback.  Ubuntu uses wx 2.8.8.0, so perhaps I'm using
something that works in 2.8.9.1 but fails for 2.8.8.0.

```
Traceback (most recent call last):
 File "/usr/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py",
line 14550, in <lambda>
   lambda event: event.callable(*event.args, **event.kw) )
 File "/home/rob/src/peppy/peppy/sidebar.py", line 86, in initPostCallback
   self.createContextMenuEventBindings()
 File "/home/rob/src/peppy/peppy/context_menu.py", line 32, in
createContextMenuEventBindings
   self.Bind(wx.EVT_CONTEXT_MENU, self.OnContextMenu)
 File "/usr/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py",
line 3886, in Bind
   event.Bind(self, id, id2, handler)
 File "/usr/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py",
line 3960, in Bind
   target.Connect(id1, id2, et, function)
 File "/usr/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py",
line 3846, in Connect
   return _core_.EvtHandler_Connect(*args, **kwargs)
TypeError: in method 'EvtHandler_Connect', expected argument 1 of type
'wxEvtHandler *'
springtabs.py:264 SpringTabItem.OnActivate: Activating File Browser:
True, shown=True
file_browser.py:47 FileBrowser.activateSpringTab:
<__builtin__.HTMLViewMode; proxy of <Swig Object of type
'wxPyHtmlWindow *' at 0xa462ca0> >
file_browser.py:49 FileBrowser.activateSpringTab: /home/rob
springtabs.py:283 SpringTabItem.setPopupFocusCallback:
springtabs.py:286 SpringTabItem.setPopupFocusCallback: setting focus
to File Browser
springtabs.py:373 SpringTabs.clearRadio: Removing popup File Browser
springtabs.py:264 SpringTabItem.OnActivate: Activating File Browser:
False, shown=False
springtabs.py:365 SpringTabs.popupItem: Popping up File Browser
springtabs.py:264 SpringTabItem.OnActivate: Activating File Browser:
True, shown=True
```

**Comment by robmcmullen** [comment]
_[Trac time 20090128 213139Z]_ Appears to be unrelated to that context menu error message, as I see that message on a working peppy.  It must be something in the GenericDirCtrl.

**Comment by robmcmullen** [comment]
_[Trac time 20090128 232858Z]_ 6e1bbe0b6ca014b7db6acea40df4e8ccd1e604c6 Fixed #584: had a call to CaptureMouse from an old attempt at getting springtabs to work
- newer versions of wxPython didn't complain about the lack of a ReleaseMouse call, but older versions (at least up to 2.8.8.0) would get stuck in an endless loop
